### PR TITLE
Add age dependancy to corona disease pathway

### PR DESCRIPTION
**Pull request by bgmoshe**

### CODE REVIEW

**Comment by bgmoshe** [comment]
@RoyIsland  @tal-nc I did not finish all of this. The part you see is mainly some code cleaning to get rid of rvs

File: src/corona_hakab_model/consts.py [hunk 1]
```diff
@@ -58,39 +58,48 @@ class Consts(NamedTuple):
     # Think we should do something more readable later on.
     # For example: "latent_to_silent_days": {"type":"uniform","lower_bound":1,"upper_bound":3}
     # disease states transition lengths distributions
-    latent_to_pre_symptomatic_days: rv_discrete = dist(1, 5, 10)
+#    latent_to_pre_symptomatic_days: BucketDict = BucketDict({
+#                                                  8: dist(1, 5, 10),  # Binomial distribution for ages<=8
+#                                                  40: dist(1, 5, 10),
+#                                                  70: dist(1, 5, 10)})
+    latent_to_pre_symptomatic_days = dist(1, 5, 10)
```
Comment (ghost) [inline src/corona_hakab_model/consts.py]: you have to add a type hint in order for the config loading to work

**Comment by tal-nc** [comment]
Overall looks good.
Just 2 points:
firstly, As Roy said, you must add type hints in order to allow loading from a file. Callable[[Optional[int]], int] should do.
secondly, why do we still need 'get_numpy_uniform_dist'? you can just implement it inside dist.

**Comment by bgmoshe** [comment]
> Overall looks good.
> Just 2 points:
> firstly, As Roy said, you must add type hints in order to allow loading from a file. Callable[[Optional[int]], int] should do.
> secondly, why do we still need 'get_numpy_uniform_dist'? you can just implement it inside dist.

You are right, I forgot about this, I'll fix this now with a small fix I need to push

File: src/corona_hakab_model/consts.py [hunk 2]
```diff
@@ -102,22 +102,24 @@ class Consts(NamedTuple):
...
-    close_medical_care_to_icu_prob: Union[float, type(...)] = 0.26
-    close_medical_care_to_mild_prob: Union[float, type(...)] = ...
-    need_icu_to_deceased_prob: Union[float, type(...)] = 0.3
-    need_icu_to_improving_prob: Union[float, type(...)] = ...
-    improving_to_need_icu_prob: Union[float, type(...)] = 0
-    improving_to_pre_recovered_prob: Union[float, type(...)] = ...
-    improving_to_mild_condition_prob: Union[float, type(...)] = 0
-    pre_recovered_to_recovered_prob: Union[float, type(...)] = ...
-    asymptomatic_end_to_recovered_prob: Union[float, type(...)] = ...
+    latent_to_latent_asymp_begin_prob: BucketDict[int, Union[float, type(...)]] = BucketDict({8: 0.3})
```
Comment (meirhalachmi) [inline src/corona_hakab_model/consts.py]: FOR HEAVEN'S SAKE - pre-define this type somewhere above and use the new alias here.

**Comment by bgmoshe** [comment]
Develop:
main

My branch:
pathway

File: src/corona_hakab_model/Parameters/simulation_parameters_example.py [hunk 3]
```diff
@@ -28,68 +28,76 @@
...
-    "close_medical_care_to_icu_prob": 0.26,
-    "close_medical_care_to_mild_end_prob": ...,
-    "need_icu_to_deceased_prob": 0.3,
-    "need_icu_to_improving_prob": ...,
-    "improving_to_need_icu_prob":0,
-    "improving_to_pre_recovered_prob": ...,
-    "improving_to_mild_condition_end_prob": 0,
-    "pre_recovered_to_recovered_prob": ...,
-    "asymptomatic_end_to_recovered_prob": ...,
+    "latent_to_latent_asymp_begin_prob": BucketDict({8: 0.3}),
```
Comment (meirhalachmi) [inline src/corona_hakab_model/Parameters/simulation_parameters_example.py:80]: The 8 does kinda confuse. Just fucking use 0 as the default key like a normal person please.

**Comment by bgmoshe** [comment]
[rvs_numpy_binom_diff.zip](https://github.com/CoronaHakab/CoronaHakabModel/files/4546603/rvs_numpy_binom_diff.zip)

**Review by meirhalachmi — CHANGES_REQUESTED** [review]

**Review by meirhalachmi — APPROVED** [review]

**Pull request merged**